server-utils/src/hosts.rs

```diff
@@ -41,45 +107,58 @@ pub fn is_host_valid(host: Option<&str>, allowed_hosts: &Option<Vec<Host>>) -> b
 		Some(ref allowed_hosts) => match host {
 			None => false,
 			Some(ref host) => {
-				allowed_hosts.iter().any(|h| h.eq_ignore_ascii_case(host) || h == host)
+				allowed_hosts.iter().any(|h| h.eq_ignore_ascii_case(host) || *host == &**h)
 			}
 		}
 	}
 }
 
-#[test]
-fn should_reject_when_there_is_no_header() {
-	let valid = is_host_valid(None, &Some(vec![]));
-	assert_eq!(valid, false);
-}
+#[cfg(test)]
+mod tests {
+	use super::{Host, is_host_valid};
 
-#[test]
-fn should_reject_when_validation_is_disabled() {
-	let valid = is_host_valid(Some("any"), &None);
-	assert_eq!(valid, true);
-}
+	#[test]
+	fn should_parse_host() {
+		assert_eq!(Host::parse("http://parity.io"), Host::new("parity.io", None));
+		assert_eq!(Host::parse("https://parity.io:8443"), Host::new("parity.io", Some(8443)));
+		assert_eq!(Host::parse("chrome-extension://124.0.0.1"), Host::new("124.0.0.1", None));
+		assert_eq!(Host::parse("parity.io/somepath"), Host::new("parity.io", None));
+		assert_eq!(Host::parse("127.0.0.1:8545/somepath"), Host::new("127.0.0.1", Some(8545)));
```

Comment:
That's the whole point.
I was considering implementation of `FromStr` instead, but then thought that it doesn't make much sense. I'm not really trying to validate if supplied hosts are actually a valid addresses (that would be much harder), but rather trying to provide a convenience method to input them, pre-processing invalid entries.